src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -493,49 +463,126 @@ def handleSuccessful(self, jobID, fwkJobReport, fwkJobReportPath = None):
                                                 conn = self.getDBConn(),
                                                 transaction = self.existingTransaction())
 
-        fileList = fwkJobReport.getAllFiles()
-        
-        bookKeepingSuccess = True
-        
-        for fwjrFile in fileList:
-            # associate logArchived file for parent jobs on wmstats assuming fileList is length is 1.
-            if jobType == "LogCollect":
+        if jobSuccess:
+            fileList = fwkJobReport.getAllFiles()
+
+            # consistency check comparing outputMap to fileList
+            # they should match except for some limited special cases
+            outputModules = set([])
+            for fwjrFile in fileList:
+                outputModules.add(fwjrFile['outputModule'])
+            if set(outputMap.keys()) == outputModules:
+                pass
+            elif jobType == "LogCollect" and len(outputMap.keys()) == 0 and outputModules == set(['LogCollect']):
+                pass
+            elif jobType == "Merge" and set(outputMap.keys()) == set(['Merged', 'MergedError', 'logArchive']) and outputModules == set(['Merged', 'logArchive']):
+                pass
+            elif jobType == "Merge" and set(outputMap.keys()) == set(['Merged', 'MergedError', 'logArchive']) and outputModules == set(['MergedError', 'logArchive']):
+                pass
+            elif jobType == "Express" and set(outputMap.keys()).difference(outputModules) == set(['write_RAW']):
+                pass
+            else:
+                failJob = True
+                if jobType in [ "Processing", "Production" ]:
+                    cmsRunSteps = 0
+                    for step in fwkJobReport.listSteps():
+                        if step.startswith("cmsRun"):
+                            cmsRunSteps += 1
+                    if cmsRunSteps > 1:
```

Comment:
Dirk, why do you fail Processing/Production jobs with > 1 cmsRuns?
I understood the idea was not to fail them, but to skip the expected outputmodules check.